core/src/test/java/org/elasticsearch/index/query/CommonTermsQueryParserTest.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.query;
+
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.test.ESSingleNodeTestCase;
+import org.junit.Test;
+
+import java.io.IOException;
+
+import static org.elasticsearch.common.xcontent.XContentFactory.jsonBuilder;
+
+public class CommonTermsQueryParserTest extends ESSingleNodeTestCase {
+    @Test
+    public void testWhenParsedQueryIsNullNoNullPointerExceptionIsThrown() throws IOException {
+        final String index = "test-index";
+        final String type = "test-type";
+        final XContentBuilder mapping =
+                jsonBuilder()
+                    .startObject()
+                        .startObject(type)
+                            .startObject("properties")
+                                .startObject("name")
+                                    .field("type", "string")
+                                    .field("analyzer", "stop")
+                                .endObject()
+                            .endObject()
+                        .endObject()
+                    .endObject();
+        client().admin().indices().prepareCreate(index).addMapping(type, mapping).execute().actionGet();
```

Comment:
I think you can shorten the mapping here: `addMapping(type, "name", "type=string, analyzer=stop"` that hurts my eyes less :)